housing_cooperative_base/views/housing.xml

```diff
@@ -36,21 +36,14 @@
                         </group>
                     </group>
                     <notebook>
-                        <page name="leases" string="Leases">
+                        <page name="rooms" string="Rooms" nolabel="1">
                             <group>
-                                <field name="lease_ids"  nolabel="1">
-                                    <tree>
-                                        <field name="tenant_id"/>
-                                        <field name="start"/>
-                                        <field name="end"/>
-                                        <field name="state"/>
-                                    </tree>
-                                </field>
+                                <field name="room_ids" options="{'no_create': True}" widget="many2many"/>
```

Comment:
If you don't set the columns, odoo will use the tree view with the highest priority, here `lease_view_tree`: 

```xml 
     <record id="lease_view_tree" model="ir.ui.view">
         <field name="name">lease_view_tree</field>
         <field name="model">hc.lease</field>
         <field name="arch" type="xml">
             <tree string="lease_tree">
                 <field name="housing_id"/>
                 <field name="start"/>
                 <field name="end"/>
                 <field name="tenant_id"/>
                 <field name="state"/>
             </tree>
         </field>
     </record>
```

Which has a useless `housing_id` column (we're on the housing form) and has an order less useful here (`tenant_id` first seems more logical to me)

Reply:
I know! There were two leases pages so I changed one to rooms (the other can be found below but was invisible is this shorter diff). Hence my changes..